oadp-operator/controllers/restic.go

```diff
@@ -1,14 +1,263 @@
 package controllers
 
 import (
+	"fmt"
+	"os"
+
 	"github.com/go-logr/logr"
 	oadpv1alpha1 "github.com/openshift/oadp-operator/api/v1alpha1"
+	"github.com/openshift/oadp-operator/pkg/credentials"
+	appsv1 "k8s.io/api/apps/v1"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/utils/pointer"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+)
+
+/**
+- name: "Enable restic"
+  k8s:
+    state: "{{ velero_state }}"
+    definition: "{{ lookup('template', 'restic.yml.j2')}}"
+  when: enable_restic == true
+*/
+
+const (
+	veleroSAName     = "velero"
+	resticPvHostPath = "/var/lib/kubelet/pods"
+)
+
+// const mountPropagationMode = v1.MountPropagationMode
+var (
+	mountPropagationToHostContainer = v1.MountPropagationHostToContainer
+	resticLabelMap                  = map[string]string{
+		"name": "restic",
+	}
 )
 
 func (r *VeleroReconciler) ReconcileResticDaemonset(log logr.Logger) (bool, error) {
 	velero := oadpv1alpha1.Velero{}
 	if err := r.Get(r.Context, r.NamespacedName, &velero); err != nil {
 		return false, err
 	}
+
+	// Define "static" portion of daemonset
+	ds := &appsv1.DaemonSet{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      resticLabelMap["name"],
```

Comment:
Not sure I understand why we are using a map for this. Can't we just declare this as a constant?